From: Dan Minor <dminor@mozilla.com>
Date: Mon, 5 Nov 2018 10:33:00 -0500
Subject: Bug 1376873 - Reduce thread stack size in platform_thread.cc; r=bwc

Summary:
The current default stack size of 1M results in intermittent OOMs on win32
builds while running web-platform tests. The value of 256k was chosen for
consistency with the default value used elsewhere in Gecko, which is defined in
nsIThreadManager.idl.

Reviewers: bwc

Tags: #secure-revision

Bug #: 1376873

Differential Revision: https://phabricator.services.mozilla.com/D11090
Mercurial Revision: https://hg.mozilla.org/mozilla-central/rev/e83c311e5293902be4db4ecea17cff87c633f7cf
---
 rtc_base/platform_thread.cc | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/rtc_base/platform_thread.cc b/rtc_base/platform_thread.cc
index 6af1f3f616..f75ce8c3dc 100644
--- a/rtc_base/platform_thread.cc
+++ b/rtc_base/platform_thread.cc
@@ -191,15 +191,17 @@ PlatformThread PlatformThread::SpawnThread(
   // Set the reserved stack stack size to 1M, which is the default on Windows
   // and Linux.
   DWORD thread_id = 0;
+  // Mozilla: Set to 256kb for consistency with nsIThreadManager.idl
   PlatformThread::Handle handle = ::CreateThread(
-      nullptr, 1024 * 1024, &RunPlatformThread, start_thread_function_ptr,
+      nullptr, 256 * 1024, &RunPlatformThread, start_thread_function_ptr,
       STACK_SIZE_PARAM_IS_A_RESERVATION, &thread_id);
   RTC_CHECK(handle) << "CreateThread failed";
 #else
   pthread_attr_t attr;
   pthread_attr_init(&attr);
   // Set the stack stack size to 1M.
-  pthread_attr_setstacksize(&attr, 1024 * 1024);
+  // Mozilla: Set to 256kb for consistency with nsIThreadManager.idl
+  pthread_attr_setstacksize(&attr, 256 * 1024);
   pthread_attr_setdetachstate(
       &attr, joinable ? PTHREAD_CREATE_JOINABLE : PTHREAD_CREATE_DETACHED);
   PlatformThread::Handle handle;
